modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1234,27 +1232,26 @@ private CompletableFuture<Void> dropTableAsyncInternal(String name) {
             if (tbl == null) {
                 dropTblFut.completeExceptionally(new TableNotFoundException(name));
             } else {
-                tablesCfg.tables()
-                        .change(change -> {
-                            TableView tableCfg = change.get(name);
+                tablesCfg.change(chg ->
+                        chg.changeTables(tblChg -> {
+                            TableView tableCfg = tblChg.get(name);
 
                             if (tableCfg == null) {
                                 throw new TableNotFoundException(name);
                             }
 
-                            NamedListView<TableIndexView> idxView = tablesCfg.indexes().value();
+                            tblChg.delete(name);
+                        }
+                        ).changeIndexes(idxChg -> {
+                            List<String> indicesNames = tablesCfg.indexes().value().namedListKeys();
 
-                            boolean idxFound = idxView.namedListKeys().stream()
-                                    .anyMatch(idx -> idxView.get(idx).tableId().equals(tbl.tableId()));
+                            List<String> indexes = indicesNames.stream().filter(idx ->
+                                    tablesCfg.indexes().get(idx).tableId().value().equals(tbl.tableId())).collect(Collectors.toList());
```

Comment:
```suggestion
                            List<String> indexes = indicesNames.stream()
                                    .filter(idx -> tablesCfg.indexes().get(idx).tableId().value().equals(tbl.tableId()))
                                    .foreach(name -> idxChg.delete(name);
```
there is no need to collect results.